device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -127,14 +133,44 @@ else if (receivePeriod == 0) //default builder value for this option, signals th
      * <p>
      * If this client is already open, then this method will do nothing.
      * <p>
-     * @throws IOException If any IO errors occur while opening the multiplexed connection.
+     * @throws MultiplexingClientException If any IO or authentication errors occur while opening the multiplexed connection.
+     * @throws MultiplexingClientDeviceRegistrationFailedException If one or many of the registered devices failed to authenticate.
```

Comment:
could there be any other reason for the device registration to fail? is it only authentication failures that would cause a `MultiplexingClientDeviceRegistrationFailedException` to be thrwon?

Reply:
I think I will rename this particular exception to ```MultiplexingClientDeviceRegistrationAuthenticationException``` and if there are other cases when the SDK needs to throw, I can add a new Exception type to cover them. Right now, this exception is only thrown in authentication failure cases.